source/MaterialXRenderGlsl/GLTextureHandler.cpp

```diff
@@ -210,18 +214,30 @@ bool GLTextureHandler::bindImage(const FilePath& filePath, const ImageSamplingPr
 
 int GLTextureHandler::mapAddressModeToGL(int addressModeEnum)
 {
-    int addressMode = GL_REPEAT;
-
-    // Mapping is from "black". Use clamp to border
-    // with border color black to achieve this
-    if (addressModeEnum == 0)
+    const vector<int> addressModes =
     {
-        addressMode = GL_CLAMP_TO_BORDER;
-    }
-    // Clamp
-    else if (addressModeEnum == 1)
+        // Mapping is from "black". Use clamp to border
+        // with border color black to achieve this
+        GL_CLAMP_TO_BORDER,
+
+        // Clamp
+        GL_CLAMP_TO_EDGE,
+
+        // Repeat
+        GL_REPEAT,
+
+        // Mirror
+        GL_MIRRORED_REPEAT,
+
+        // Mapping is from "default". Use clamp to border
+        // with default color to achieve this
+        GL_CLAMP_TO_BORDER
+    };
+
+    int addressMode = GL_REPEAT;
+    if (addressModeEnum >= 0 && addressModeEnum < 5)
```

Comment:
Maybe the MaterialX adressmodes should be enums to increase code readability.

Reply:
Yes. There is a bit of "magic" numbering here. I'll fix.